tests/Unit/ParallelAlgorithms/Events/Test_ObserveFields.cpp

```diff
@@ -395,27 +416,96 @@ void test_observe(const std::unique_ptr<ObserveEvent> observe) noexcept {
 }
 
 template <typename System>
-void test_system() noexcept {
+void test_system(const std::string& mesh_creation_string,
+                 const std::optional<Mesh<System::volume_dim>>&
+                     interpolating_mesh = {}) noexcept {
   INFO(pretty_type::get_name<System>());
   test_observe<System>(std::make_unique<typename System::ObserveEvent>(
-      System::make_test_object()));
+                           System::make_test_object(interpolating_mesh)),
+                       interpolating_mesh);
 
   INFO("create/serialize");
   using EventType = Event<tmpl::list<dg::Events::Registrars::ObserveFields<
       System::volume_dim, ObservationTimeTag,
       typename System::all_vars_for_test,
       typename System::solution_for_test::vars_for_test>>>;
   Parallel::register_derived_classes_with_charm<EventType>();
-  const auto factory_event = TestHelpers::test_factory_creation<EventType>(
-      System::creation_string_for_test);
+  const std::string creation_string =
+      System::creation_string_for_test + mesh_creation_string;
+  const auto factory_event =
+      TestHelpers::test_factory_creation<EventType>(creation_string);
   auto serialized_event = serialize_and_deserialize(factory_event);
-  test_observe<System>(std::move(serialized_event));
+  test_observe<System>(std::move(serialized_event), interpolating_mesh);
 }
 }  // namespace
 
 SPECTRE_TEST_CASE("Unit.Evolution.dG.ObserveFields", "[Unit][Evolution]") {
-  test_system<ScalarSystem>();
-  test_system<ComplicatedSystem>();
+  const std::string no_mesh_str = "  InterpolateToMesh: None";
+  test_system<ScalarSystem>(no_mesh_str);
+  test_system<ComplicatedSystem>(no_mesh_str);
+
+  const std::string more_points_mesh_str =
+      "  InterpolateToMesh:\n"
+      "    Extents: 12\n"
+      "    Basis: Legendre\n"
+      "    Quadrature: GaussLobatto";
+  const Mesh<1> more_points_mesh_1{12, Spectral::Basis::Legendre,
+                                   Spectral::Quadrature::GaussLobatto};
+  const Mesh<2> more_points_mesh_2{12, Spectral::Basis::Legendre,
+                                   Spectral::Quadrature::GaussLobatto};
```

Comment:
Test nonuniform meshes.  We can't parse them yet, but we don't want to have to change this class when we can.